sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -325,6 +326,28 @@ object WholeStageCodegenExec {
   }
 }
 
+object WholeStageCodegenId {
+  private val codegenStageCounter = ThreadLocal.withInitial(new Supplier[Integer] {
+    override def get() = 0  // TODO: change to Scala lambda syntax when upgraded to Scala 2.12+
```

Comment:
shall we just use 1 as initial value and add a comment to say that 0 is preserved for temporary WholeStageCodegenExec objects? Then we only need a `reset` method.

Reply:
With the updated PR that uses the secondary constructor in `WholeStageCodegenExec`, yes you're making a good point. All the places that create temporary `WholeStageCodegenExec` objects are explicitly passing in `0` as the codegen stage ID now, so we can indeed simplify the counter logic here.

Will address in the next update.